File: flink-java-code/src/main/java/com/atguigu/day02/FilterFunctionExample.java
package com.atguigu.day02;

import org.apache.flink.api.common.functions.FilterFunction;
import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

import java.util.logging.Filter;

public class FilterFunctionExample {
    public static void main(String[] args) throws Exception{
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);

        DataStreamSource<SensorReading> stream = env.addSource(new SensorSource());
        stream.filter(f -> f.temperature > 0);

        stream.filter(new MyFliterFilFunction());

        stream.filter(new FilterFunction<SensorReading>() {
            @Override
            public boolean filter(SensorReading sensorReading) throws Exception {
                return sensorReading.temperature > 0;
            }
        }).print();
        env.execute();


    }
    public static class MyFliterFilFunction implements FilterFunction<SensorReading>
    {
        @Override
        public boolean filter(SensorReading sensorReading) throws Exception {
            return sensorReading.temperature>0;
        }
    }
}

File: FlinkTutorial/src/main/scala/com/atguigu/day02/FilterFunctionExample.scala
package com.atguigu.day02

import org.apache.flink.api.common.functions.{FilterFunction, MapFunction}
import org.apache.flink.streaming.api.scala._

object FilterFunctionExample {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(1)
    val stream = env.addSource(new SensorSource)

    stream.filter(t => t.temaperature>0)

    stream.filter(new MyFilterFunction)

    stream.filter(new FilterFunction[SensorReading] {
      override def filter(t: SensorReading): Boolean = t.temaperature > 0
    }).print()

    env.execute()
  }
  class MyFilterFunction extends FilterFunction[SensorReading] {
    override def filter(t: SensorReading): Boolean = t.temaperature > 0
  }
}
